publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/expandresources/ExpandedResourcePersistenceHandler.java

```diff
@@ -43,7 +44,17 @@ protected EventPayload processInputPayload(EventPayload input,
             attempt(() -> dynamoImageSerializerRemovingEmptyFields.readValue(data, ExpandedDatabaseEntry.class))
                 .orElseThrow();
         var indexDocument = createIndexDocument(expandedResourceUpdate);
-        URI uri = attempt(() -> s3Writer.insertEvent(UnixPath.EMPTY_PATH, indexDocument.toJsonString())).orElseThrow();
+        var filePath = createFilePath(indexDocument);
+        URI uri = attempt(() -> s3Writer.insertFile(filePath, indexDocument.toJsonString())).orElseThrow();
         return EventPayload.indexEntryEvent(uri);
     }
+
+    private UnixPath createFilePath(PersistedDocument indexDocument) {
+        return UnixPath.of(createPathBasedOnIndexName(indexDocument))
+            .addChild(indexDocument.getMetadata().getDocumentIdentifier().toString() + GZIP_ENDING);
+    }
+
+    private String createPathBasedOnIndexName(PersistedDocument indexDocument) {
```

Comment:
It seems a bit wrong that the persistence of data should be affected by the es indices but i think that the index name is just a resource type.